app/src/main/java/com/duckduckgo/app/browser/favicon/FaviconManager.kt

```diff
@@ -64,13 +66,23 @@ class DuckDuckGoFaviconManager constructor(
     }
 
     override suspend fun loadToViewFromPersisted(url: String, view: ImageView) {
+        // avoid displaying the previous favicon when the holder is recycled
+        preloadDefaultImageIntoView(view)
         loadToViewFromDirectory(url, FAVICON_PERSISTED_DIR, NO_SUBFOLDER, view)
     }
 
     override suspend fun loadToViewFromTemp(subFolder: String, url: String, view: ImageView) {
+        // avoid displaying the previous favicon when the holder is recycled
+        preloadDefaultImageIntoView(view)
         loadToViewFromDirectory(url, FAVICON_TEMP_DIR, subFolder, view)
     }
 
+    private suspend fun preloadDefaultImageIntoView(view: ImageView) {
```

Comment:
This looks great. Just one proposal from my side. Since the implementation detail of what placeholder we use for favicons (`ic_globe_gray_16dp`) is inside the `faviconDownloader`, just to keep any similar logic in one place, will it be possible to move this method inside the `FaviconDownloader`?

Reply:
Makes sense, will do

Reply:
@cmonfortep addressed the comment. Let me know any other comments you might have. Feel free to resolve this conversation if all is OK